kernel/src/fs/ramfs.rs

```
@@ -95,12 +95,17 @@ impl RawRamFile {
         let page_offset = page_offset(file_offset);
         let page_index = file_offset / PAGE_SIZE;
 
+        // Make sure the page exists before trying to read from it.
+        let Some(page) = self.pages.get(page_index) else {
+            return Ok(0);
```

Should we return an error on this case?
Or at least document this behavior.

---

No, this is just a read attempt beyond EOF, which usually returns `0-bytes-read`

---

I see, I would document that, but for sure not in this PR.